lib/ui/pages/workspace_page/components/drawing_surface/canvas_painter.dart

```diff
@@ -57,6 +59,8 @@ class PaintingLayer extends ConsumerWidget {
     ref.watch(commandManagerProvider);
     ref.watch(canvasPainterProvider);
     ref.watch(paintProvider);
+    ref.watch(toolBoxStateProvider);
+    ref.watch(textToolProvider);
```

Comment:
Generally no need for `ref.watch(textToolProvider);` You can use the `.repaint()`side effect to trigger a rebuild using `CanvasPainterProvider`